pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JModifier.java

```diff
@@ -26,27 +26,25 @@ public enum JModifier {
     PROTECTED(Modifier.PROTECTED),
     PRIVATE(Modifier.PRIVATE),
 
+    ABSTRACT(Modifier.ABSTRACT),
+    DEFAULT(0),
+    STATIC(Modifier.STATIC),
+    FINAL(Modifier.FINAL),
+
     /** Modifier {@code "sealed"} (since Java 17). */
     SEALED(0),
     /** Modifier {@code "non-sealed"} (since Java 17). */
     NON_SEALED("non-sealed", 0),
 
-    ABSTRACT(Modifier.ABSTRACT),
-    STATIC(Modifier.STATIC),
-    FINAL(Modifier.FINAL),
-
+    // for fields
```

Comment:
I think these comments should be removed. They're likely the reason the modifiers ended up in the wrong order: I added constants in one of those sections instead of focusing on the ordering

Reply:
Maybe even a comment should be added to say something like 
```java
/*******************************************************************
    NOTE: When adding a constant, check the ordering is consistent with the recommended modifier ordering!
********************************************************************/
```
or something similarly "high visibility"